model-optimizer/extensions/middle/MarkSubgraphsWithCorrectLayout.py

```diff
@@ -186,22 +255,29 @@ def get_ports_and_nodes_on_weights(graph):
 
         # collect all input ports with weights
         weight_ports = set()
+        result_ports = set()
         start_ports = set()
         for node in nodes:
             node_type = node.soft_get('type', 'unknown')
             if node_type not in weighted_types:
-                if node_type in ['Const', 'Parameter', 'ShapeOf']:
+                if node_type in ['Const', 'Parameter', 'ShapeOf', 'ExtractImagePatches']:
```

Comment:
?! elaborate more on this change

Reply:
Yes, so I decided to make ExtractImagePatches as weight path producer. Meaning that if we meat EIP node we make it a start of weight path

Reply:
It's not true. Why would we do that?

Reply:
Why not? When we go up from the weight input of Convolution and meet EIP node, in my opinion, it is safe to treat EIP as the start of the weight path, why do you think it is wrong?

Reply:
I agree that the cannot blindly consider "EIP" as a weight path producer. @mvafin , have you analyzed other models with the EIP operation? For example, Yolo models. I want to understand what will happen if the EIP is not on a weight path? Will MO incorrectly convert the model?

Also I see that the MO tests fails. It should also be investigated.